### ISSUE TITLE
No "exports" main resolved in project/functions/my-function/node_modules/faunadb/package.json

### ISSUE THREAD

**Issue by JonasTaulienSolutions** [netlify/cli#893]
**- Do you want to request a _feature_ or report a _bug_?**
I want to report a behaviour that seems like a bug to me. If it is not, I really would appreciate help about what I did wrong.

**- What is the current behavior?**
My project has the following structure:
* `project/`
    * `netlify-functions/`
        * `my-function/`
            * `node_modules/`
            * `my-function.js`
            * `package.json`
    * `src/`
    * `netlify.toml`
    * `package.json`

Content of `netlify.toml`:
```
[build]
  base      = ""
  publish   = "dist"
  command   = "yarn build"
  functions = "netlify-functions/"

# ...

[dev]
  command = "yarn lite-server --config=lite-server.config.json"
  framework = "#custom"
  port = 8880
  targetPort = 3000
  autoLaunch = true
```

Content of `project/package.json`:
```
{
 // ...
  "scripts": {
    // ...
    "build": "webpack",
    "serve": "netlify dev"
  },
  "dependencies": {
    // ...
    "netlify-cli": "^2.50.0"
  }
}
```

Content of `project/netlify-functions/my-function/package.json`:
```
{
  // ...
  "dependencies": {
    "faunadb": "^2.14.0"
  }
}
```

Content of `project/netlify-functions/my-function/my-function.js`:
```
const faunadb = require('faunadb');
// ...
exports.handler = async function (event, context) {
    // ...
};
```

**- If the current behavior is a bug, please provide the steps to reproduce.**
1. Execute `yarn serve` in a project with the above setup

    ```
    >>> yarn serve
    yarn run v1.22.4
    $ netlify dev
    ◈ Netlify Dev ◈
    ◈ Injected addon env var:  FAUNADB_ADMIN_SECRET
    ◈ Injected addon env var:  FAUNADB_SERVER_SECRET
    ◈ Injected addon env var:  FAUNADB_CLIENT_SECRET
    ◈ Starting Netlify Dev with #custom
    Waiting for localhost:3000.
    ◈ Functions server is listening on 34567
    $ ... /node_modules/.bin/lite-server --config=lite-server.config.json
    ** browser-sync config **
    {
      injectChanges: false,
      files: [ './**/*.{html,htm,css,js}' ],
      watchOptions: { ignored: 'node_modules' },
      server: {
        baseDir: 'dist',
        middleware: [ [Function (anonymous)], [Function (anonymous)] ]
      },
      open: false
    }
    [Browsersync] Access URLs:
     ---------------------------------------
           Local: http://localhost:3000
        External: http://192.168.178.33:3000
     ---------------------------------------
              UI: http://localhost:3001
     UI External: http://localhost:3001
     ---------------------------------------
    [.
    Connected!
    Waiting for localhost:34567.
    Connected!
    
       ┌─────────────────────────────────────────────────┐
       │                                                 │
       │   ◈ Server now ready on http://localhost:8880   │
       │                                                 │
       └─────────────────────────────────────────────────┘
    
    Browsersync] Serving files from: dist
    [Browsersync] Watching files...
    20.05.11 20:32:57 304 GET /index.html
    20.05.11 20:32:57 304 
    ```

2. Call `my-function`

The output is:
```
20.05.11 20:32:57 304 Request from ::1: POST /.netlify/functions/my-function
{"level":"error","message":"End - Error:"}
{"errorMessage":"No \"exports\" main resolved in project/netlify-functions/my-function/node_modules/faunadb/package.json","errorType":"Error","stackTrace":["resolveExportsTarget (internal/modules/cjs/loader.js:628:11)","applyExports (internal/modules/cjs/loader.js:503:14)","resolveExports (internal/modules/cjs/loader.js:541:12)","Function.Module._findPath (internal/modules/cjs/loader.js:661:22)","Function.Module._resolveFilename (internal/modules/cjs/loader.js:963:27)","Function.Module._load (internal/modules/cjs/loader.js:859:27)","Module.require (internal/modules/cjs/loader.js:1036:19)","require (internal/modules/cjs/helpers.js:72:18)","Object.<anonymous> (project/netlify-functions/my-function/my-function.js:1:17)","Module._compile (internal/modules/cjs/loader.js:1147:30)"],"level":"error"}
Response with status 500 in 26 ms.
◈ Error during invocation:  {
  errorMessage: 'No "exports" main resolved in project/netlify-functions/my-function/node_modules/faunadb/package.json',
  errorType: 'Error',
  stackTrace: [
    'resolveExportsTarget (internal/modules/cjs/loader.js:628:11)',
    'applyExports (internal/modules/cjs/loader.js:503:14)',
    'resolveExports (internal/modules/cjs/loader.js:541:12)',
    'Function.Module._findPath (internal/modules/cjs/loader.js:661:22)',
    'Function.Module._resolveFilename (internal/modules/cjs/loader.js:963:27)',
    'Function.Module._load (internal/modules/cjs/loader.js:859:27)',
    'Module.require (internal/modules/cjs/loader.js:1036:19)',
    'require (internal/modules/cjs/helpers.js:72:18)',
    'Object.<anonymous> (project/netlify-functions/my-function/my-function.js:1:17)',
    'Module._compile (internal/modules/cjs/loader.js:1147:30)'
  ],
  level: 'error',
  [Symbol(level)]: 'error',
  [Symbol(message)]: '{"errorMessage":"No \\"exports\\" main resolved in project/netlify-functions/my-function/node_modules/faunadb/package.json","errorType":"Error","stackTrace":["resolveExportsTarget (internal/modules/cjs/loader.js:628:11)","applyExports (internal/modules/cjs/loader.js:503:14)","resolveExports (internal/modules/cjs/loader.js:541:12)","Function.Module._findPath (internal/modules/cjs/loader.js:661:22)","Function.Module._resolveFilename (internal/modules/cjs/loader.js:963:27)","Function.Module._load (internal/modules/cjs/loader.js:859:27)","Module.require (internal/modules/cjs/loader.js:1036:19)","require (internal/modules/cjs/helpers.js:72:18)","Object.<anonymous> (project/netlify-functions/my-function/my-function.js:1:17)","Module._compile (internal/modules/cjs/loader.js:1147:30)"],"level":"error"}'
}

```

**- What is the expected behavior?**
I would expect that no error gets thrown.

**- Local Environment Information**
```
$ netlify version --verbose
────────────────────┐
 Environment Info   │
────────────────────┘

  System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
  Binaries:
    Node: 13.11.0 - /usr/local/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 6.13.7 - /usr/local/bin/npm
  Browsers:
    Chrome: 81.0.4044.138
    Firefox: 75.0
    Safari: 13.1

netlify-cli/2.50.0 darwin-x64 node-v13.11.0
```

**Comment by JonasTaulienSolutions** [comment]
Ok, acccording to Ryan Harris of Fauna this is a bug in `"faunadb": "2.14.0"`. The temporarily fix is to downgrade to `"faunadb": "2.13.1"`

**Comment by FelDev** [comment]
Seems to be patched in 2.14.2

**Comment by JonasTaulienSolutions** [comment]
Thanks :)
